ff/src/fields/arithmetic.rs

```diff
@@ -80,22 +94,28 @@ macro_rules! impl_field_into_repr {
 }
 
 macro_rules! impl_field_square_in_place {
+    (1) => {
+        fn square_in_place(&mut self) -> &mut Self {
+            *self = *self * &self;
+            self
+        }
+    };
     ($limbs: expr) => {
         #[inline]
         #[ark_ff_asm::unroll_for_loops]
         #[allow(unused_braces, clippy::absurd_extreme_comparisons)]
         fn square_in_place(&mut self) -> &mut Self {
-            // Checking the modulus at compile time
-            let first_bit_set = P::MODULUS.0[$limbs - 1] >> 63 != 0;
-            let mut all_bits_set = P::MODULUS.0[$limbs - 1] == !0 - (1 << 63);
-            for i in 1..$limbs {
-                all_bits_set &= P::MODULUS.0[$limbs - i - 1] == core::u64::MAX;
-            }
-            let _no_carry: bool = !(first_bit_set || all_bits_set);
-
             #[cfg(use_asm)]
             #[allow(unsafe_code, unused_mut)]
             {
+                // Checking the modulus at compile time
```

Comment:
Drive-by refactor that moves code used only when `use_asm` is true inside the conditionally-compiled portion.